File: algorithms/implementation/sock-merchant/sock-merchant-javascript/js/test_sockMerchant.js
import {sockMerchant} from '../js/sockMerchant.js';
import assert from 'assert';

describe('sockMerchant', function() {

    it('O número de pares é 3', function() {

        let n = 9;
        
        let ar = [10, 20, 20, 10, 10, 30, 50, 10, 20]

        let esperado = 3;

        let resultado = sockMerchant(n, ar);

        assert.strictEqual(resultado, esperado);


    });
});

describe('sockMerchant', function() {

    it('O número de pares é 4', function() {

        let n = 10;
        
        let ar = [1, 1, 3, 1, 2, 1, 3, 3, 3, 3]

        let esperado = 4;

        let resultado = sockMerchant(n, ar);

        assert.strictEqual(resultado, esperado);


    });
});

describe('sockMerchant', function() {

    it('O número de pares é 6', function() {

        let n = 15;
        
        let ar = [6, 5, 2, 3, 5, 2, 2, 1, 1, 5, 1, 3, 3, 3, 5]

        let esperado = 6;

        let resultado = sockMerchant(n, ar);

        assert.strictEqual(resultado, esperado);


    });
});

describe('sockMerchant', function() {

    it('O número de pares é 9', function() {

        let n = 20;
        
        let ar = [4, 5, 5, 5, 6, 6, 4, 1, 4, 4, 3, 6, 6, 3, 6, 1, 4, 5, 5, 5]

        let esperado = 9;

        let resultado = sockMerchant(n, ar);

        assert.strictEqual(resultado, esperado);


    });
});

File: algorithms/implementation/sock-merchant/sock-merchant-python/tests/test_sockMerchant.py
from sockMerchant import sockMerchant

class TesteSockMerchant:
    
    def testeCaso1(self):
        
        n = 9
        ar = [10, 20, 20, 10, 10, 30, 50, 10, 20]
        esperado = 3
        
        resultado = sockMerchant(n, ar)
        
        assert resultado == esperado
        
    def testeCaso2(self):
        
        n = 10
        ar = [1, 1, 3, 1, 2, 1, 3, 3, 3, 3]
        esperado = 4
        
        resultado = sockMerchant(n, ar)
        
        assert resultado == esperado
        
    def testeCaso3(self):
        
        n = 15
        ar = [6, 5, 2, 3, 5, 2, 2, 1, 1, 5, 1, 3, 3, 3, 5]
        esperado = 6
        
        resultado = sockMerchant(n, ar)
        
        assert resultado == esperado
        
    def testeCaso4(self):
        
        n = 20
        ar = [4, 5, 5, 5, 6, 6, 4, 1, 4, 4, 3, 6, 6, 3, 6, 1, 4, 5, 5, 5]
        esperado = 9
        
        resultado = sockMerchant(n, ar)
        
        assert resultado == esperado